Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -138,4 +140,40 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ScrollTimeline::documentW
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+ScrollableArea* ScrollTimeline::scrollableAreaForSourceRenderer(RenderElement* renderer, Ref<Document> document)
+{
+    CheckedPtr renderBox = dynamicDowncast<RenderBox>(renderer);
+    if (!renderBox)
+        return nullptr;
+
+    if (renderer->element() == document->documentElement())
+        return &renderer->view().frameView();
+    return (renderBox->canBeScrolledAndHasScrollableArea() && renderBox->layer()) ? renderBox->layer()->scrollableArea() : nullptr;
```

Comment:
Nit: `box->hasLayer()` is a bit check, and possibly slightly faster than checking `box->layer()`

Reply:
Fixed.